readdir.go

```diff
@@ -116,9 +116,7 @@ func (p *Path) ReadDirRecursiveFiltered(recursionFilter ReadDirFilter, filters .
 			}
 
 			if recursionFilter == nil || recursionFilter(path) {
-				if isDir, err := path.IsDirCheck(); err != nil {
-					return nil, err
-				} else if isDir {
+				if path.IsDir() {
```

Comment:
Maybe we should make this check more strict?

Reply:
I've tried some other combinations of `IsDir`, `Stat` , and `Lstat`: this one seems the most reliable.